grails-app/services/org/pih/warehouse/inventory/StockMovementService.groovy

```diff
@@ -1406,6 +1403,16 @@ class StockMovementService {
         }
     }
 
+    void updateInventoryItems(StockMovement stockMovement) {
+        if (stockMovement.lineItems) {
+            stockMovement.lineItems.each { StockMovementItem stockMovementItem ->
+                InventoryItem inventoryItem = inventoryService.findOrCreateInventoryItem(stockMovementItem.product,
```

Comment:
You could use findOrCreateInventoryItemDuringImport (perhaps you can rename it to something more generic)